src/resolver/resolver.zig

```diff
@@ -1855,7 +1855,9 @@ pub const Resolver = struct {
         if (needs_iter) {
             const allocator = r.fs.allocator;
             dir_entries_option = rfs.entries.put(&cached_dir_entry_result, .{
-                .entries = Fs.FileSystem.DirEntry.init(dir_path),
+                .entries = Fs.FileSystem.DirEntry.init(
+                    Fs.FileSystem.DirnameStore.instance.append(string, dir_path) catch unreachable,
```

Comment:
I think there are few cases where we need to clone this. The caller already cloned it except for the case you're running into, which has to do with automatic package installs

Reply:
Okay let me look into that a bit more :+1:

Reply:
We could add a debug assertion which checks:

```zig
Fs.FileSystem.DirnameStore.instance.isSliceInBuffer(dir_path) or bun.isHeapMemory(dir_path)
```

Then we can find the callers of this function which are not cloning the dir name

Reply:
There is only one user for this function:
https://github.com/oven-sh/bun/blob/30e82c5df42fab71a238cd9b7d268cbb6510bd7a/src/resolver/resolver.zig#L1717
`string` of which is supplied by `PackageManager.pathForResolution()`:
https://github.com/oven-sh/bun/blob/30e82c5df42fab71a238cd9b7d268cbb6510bd7a/src/resolver/resolver.zig#L1662
which is using a `threadlocal` buffer:
https://github.com/oven-sh/bun/blob/30e82c5df42fab71a238cd9b7d268cbb6510bd7a/src/resolver/resolver.zig#L260
... doesn't look very cloned to me :sweat_smile: